### ISSUE TITLE
default_scope generating a field with a complex query

### ISSUE THREAD

**Issue by fbernier** [mongoid/mongoid#3723]
If we have the following default scope:

```ruby
default_scope ->() { self.or({:deactivated_at => nil}, {:deactivated_at => Time.at(0)}) }
```

A record like this one is generated when calling Model.new:

```
[20] app »  Connector.new
=> #<Connector:0x007fbf4d80e148> {
  :$or => [
    [0] {
      "deactivated_at" => nil
    },
    [1] {
      "deactivated_at" => 1970-01-01 00:00:00 UTC
    }
  ],
  :_id => BSON::ObjectId('53bb042c6672611f6e090000')
}
```

What makes it worse is it's generating a field starting with a $ which raises an error on mongodb 2.6, but will pass on mongodb 2.4. This created lots of fields which are not valid for mongodb 2.6 preventing an upgrade until those fields are deleted.

See 
https://github.com/mongoid/mongoid/blob/master/lib/mongoid/scopable.rb#L20-L21

**Comment by fbernier** [comment]
Hit this problem once again today. Hopefully I'll find some time to make a fix.

**Comment by arthurnn** [comment]
@fbernier fell free to send a PR, or even a broken test case so we can simulate and easily fix it.
thanks

**Comment by fbernier** [comment]
https://github.com/mongoid/mongoid/pull/4011

**Comment by andresilveira** [comment]
Hey guys, is this patch coming down to the version 4.x ?

**Comment by andresilveira** [comment]
I've applied the same solution for the 4.0.2 release. https://github.com/andresilveira/mongoid/tree/patched-4.0.2
Should I issue a pull request?
